### ISSUE TITLE
Doubt early stopping for TableNet?

### ISSUE THREAD

**Issue by PayneJoe** [microsoft/qlib#263]
----------------- SHOW THE CODE ------------------------------

Early stopping logic of TableNet:

`            if val_score < best_score:
                best_score = val_score
                stop_steps = 0
                best_epoch = epoch_idx
            else:
                stop_steps += 1
                if stop_steps >= self.early_stop:
                    self.logger.info("early stop")
                    break
`
Early stopping for other DNNs:

`
            if val_score > best_score:
                best_score = val_score
                stop_steps = 0
                best_epoch = step
                best_param = copy.deepcopy(self.GAT_model.state_dict())
            else:
                stop_steps += 1
                if stop_steps >= self.early_stop:
                    self.logger.info("early stop")
                    break
`

Obviously, the diff is the comparison between val_score and best_score. While the default score is the negative of MSE loss, the latter one(val_score > best_score) should be right, am I right? Or do I miss some thing else?

**Comment by javaThonc** [comment]
Hi PayneJoe, thanks for your close review. You are right; In this case, the model will be stopped at the wrong epoch. We will fix it and pr a corrected version of TabNet.
